integration/fake/fake_trilian_log.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fake
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/trillian"
+)
+
+// FakeClient implements trillian/client.LogVerifier
+type FakeClient struct {
+	leaves []*trillian.LogLeaf
+}
+
+// NewFakeTrillianClient returns a client that mimicks a trillian log.
+func NewFakeTrillianClient() *FakeClient {
+	return &FakeClient{
+		leaves: make([]*trillian.LogLeaf, 0),
+	}
+}
+
+// AddLeaf adds a leaf to the log.
+func (f *FakeClient) AddLeaf(ctx context.Context, data []byte) error {
+	f.leaves = append(f.leaves, &trillian.LogLeaf{
+		LeafValue: data,
+	})
+	return nil
+}
+
+// GetByIndex returns the requested leaf.
+func (f *FakeClient) GetByIndex(ctx context.Context, index int64) (*trillian.LogLeaf, error) {
+	if got, want := index, int64(len(f.leaves)); got > want {
```

Comment:
Check my reasoning here:
- `len(f.leaves)` will be >= 0
- `int64(len(f.leaves))` upcasts length from `int` to `int64` to make sure that we're comparing to items of the same type.

Therefore a negative index will always fail this test.